ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh

```diff
@@ -6,6 +6,141 @@ set -o pipefail
 
 trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
 
+trap 'FRC=$?; createHeterogeneousJunit; debug' EXIT TERM
+
+# Print failed node, co information for debug purpose
+function debug() {
+    if (( FRC != 0 )); then
+        echo -e "Describing abnormal nodes...\n"
+        oc get node --no-headers | awk '$2 != "Ready" {print $1}' | while read node; do echo -e "\n#####oc describe node ${node}#####\n$(oc describe node ${node})"; done
+        echo -e "Describing abnormal operators...\n"
+        oc get co --no-headers | awk '$3 != "True" || $4 != "False" || $5 != "False" {print $1}' | while read co; do echo -e "\n#####oc describe co ${co}#####\n$(oc describe co ${co})"; done
+    fi
+}
+
+# Generate the Junit for migration
+function createHeterogeneousJunit() {
+    echo "Generating the Junit for heterogeneous"
+    if (( FRC == 0 )); then
+      cat >"${ARTIFACT_DIR}/junit_heterogeneous.xml" <<EOF
```

Comment:
due to the current implementation of the classifier, let's store the file as `junit_upgrade.xml`

Reply:
checked with @shellyyang1989 privately, if we store the file as juint_upgrade.xml, we might override the upgrade step junit. if we need add new classifier for junit_heterogeneous.xml?  cc @JianLi-RH

Reply:
@lwan-wanglin We read all xml files from Prow step folder to upload to ReportPortal. So both junit_upgrade.xml and junit_heterogeneous.xml are valid for me. 

If step ipi-install-heterogeneous do same work (upgrade work) as junit_upgrade in other steps (like cucushift-upgrade-toimage), I think you also can use junit_upgrade.xml here. But if not, please keep junit_heterogeneous.xml  as your output.

If the result of step ipi-install-heterogeneous  indicates upgrade result, we should update our uploading process and @pruan-rht need to update below script as well:
https://gitlab.cee.redhat.com/aosqe/prow-test-results-classfier/-/blob/main/libs/save_results.py#L151

Reply:
we don't care about the junit file if you add step ipi-install-heterogeneous into https://gitlab.cee.redhat.com/aosqe/prow-test-results-classfier/-/blob/main/libs/job.py#L131, otherwise we will read the junit to upload result to ReportPortal.

Reply:
thanks Jian for your reply. 

@aleskandro for this step, do we want to add the result to `cluster upgrade` classifier in the reportportal? if we do, Jian will help us add the step to above script, then we even don't need this junit file, if the step is failed, the script will recognize it and upload the failed logs to RP , and also it can be noticed by the upgrade team monitoring. the only issue is the step will fail not always  by arch-migration, sometimes by node scaleup issue, then it's not belong upgrade issue. the other suggestion is we create the junit, then it will create a separate classifier according to `testsuite name` in junit.  which one do we prefer?

Reply:
Hi @JianLi-RH I wanted to write you yesterday about this and other related things.

I wanted to have the name of the file set to `junit_upgrade.xml` to reduce the changes requirements in https://gitlab.cee.redhat.com/aosqe/prow-test-results-classfier/-/blob/main/libs/save_results.py#L151

And yes, we should also add this step to the list in https://gitlab.cee.redhat.com/aosqe/prow-test-results-classfier/-/blob/main/libs/job.py#L131

If this step fails, we should also consider failed the migration to the multiarch payload, although the relevant fix could land on another component. I think this result  should still be classified as a upgrade (read migration) test result.... @shellyyang1989, wdyt?


Finally, @JianLi-RH, I would reach at you offline about the need to opt-in steps into those lists in the Job-specific python classes of the tests-results classifier. I wonder if we can better balance the opt-in vs opt-out somehow to run a catch-all test-case that goes beyond the multi-arch issue. See https://issues.redhat.com/browse/OCPQE-16721

Reply:
Hi @aleskandro 
1. we can use anything as the junit file name
2.  I am afraid https://gitlab.cee.redhat.com/aosqe/prow-test-results-classfier/-/blob/main/libs/save_results.py#L151 does not work now, since we are grouping all upgrade steps into `cluster upgrade`, look at https://gitlab.cee.redhat.com/aosqe/prow-test-results-classfier/-/blob/main/libs/job.py#L543-544, you can see we are generate a new junit file which name is same with the group name.
3. for `non-upgrade jobs`, I suggest to add another group which only include the specified steps.
4. I am not quite understand your last question, feel free to ping me
/cc @lwan-wanglin